### PR TITLE
Tweak Python icons

### PR DESCRIPTION
**Pull request by mataha**
#### Description

This commit:

 - adds PyPI icon to related files (`requirements.in`, `requirements.txt`)
 - adds a modified Python icon to `py.typed` files
 - adds an icon separate from Python to `.python-version` files

#### Motivation and Context

Improved visual grepping in Python projects.

#### How Has This Been Tested?

- `py.typed`
- `requirements.in`
- `requirements.txt`

#### Types of changes:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: icons/icons.json [hunk 1]
```diff
@@ -2576,16 +2576,20 @@
...
+	"file_type_pypi": {
 		"aliases": [
 			{
 				"extensions": [
-					".python-version"
+					"requirements.in",
+					"requirements.txt"
 				],
-				"name": "Plain Text (Python)",
-				"scope": "text.plain.python"
```
Comment (deathaxe) [inline icons/icons.json:2586]: At least https://packagecontrol.io/packages/requirementstxt provides a real syntax for this file type. So maybe adjust declaration accordingly.

Also I am not quite sure requirements.txt is directly related with pypi. It's rather a way to list dependencies to be installed via pip.

see: https://pip.pypa.io/en/stable/reference/requirements-file-format/

File: icons/icons.json [hunk 2]
```diff
@@ -2594,6 +2598,30 @@
...
+		"color": "blue"
+	},
+	"file_type_python_version": {
+		"aliases": [
+			{
+				"extensions": [
+					".python-version"
+				],
+				"name": "Plain Text (Python)",
+				"scope": "text.plain.python"
```
Comment (deathaxe) [inline icons/icons.json:2620]: Two definitions with same scope doesn't work. Also do we really need dedicated icons for every python related meta-data file?

**Review by deathaxe — CHANGES_REQUESTED** [review]

File: icons/icons.json (hunk 2 above)
Comment (mataha) [reply icons/icons.json:2621]: I certainly do not think so, but I wouldn't consider these "every python related meta-data file" - just the bare minimum expected in a reasonably-sized Python project **and** tied to the tools included with the runtime.

Comment (mataha) [reply icons/icons.json:2621]: Changed to extend `source.python` instead.

File: icons/icons.json (hunk 1 above)
Comment (mataha) [reply icons/icons.json]: I didn't know about that package - adjusted the definition to use `source.requirementstxt`.

Re: PyPI - what I've aimed for here is consistency with JS filetypes - there, `package.json` is under the `npm` umbrella even though it's not tied to any package manager or package registry (or even runtime - even Deno takes `package.json` into consideration nowadays).

Changed to `file_type_pip` which I've probably should've done from the start.

File: icons/icons.json [hunk 3]
```diff
@@ -2300,6 +2300,19 @@
 			}
 		]
 	},
+	"file_type_pip": {
+		"color": "sky",
+		"syntaxes": [
+			{
+				"extensions": [
+					"requirements.in",
+					"requirements.txt"
```
[the remark this answers is not in the document]
Comment (mataha) [reply icons/icons.json:2309]: I wish this was standardised, but instead we have:

- `*-requirements.txt` ([flake8](https://github.com/PyCQA/flake8), [mypy](https://github.com/python/mypy), [urllib3](https://github.com/urllib3/urllib3))
- `*_requirements.txt` ([black](https://github.com/psf/black), [numpy](https://github.com/numpy/numpy))
- `requirements-*.txt` ([Chalice](https://github.com/aws/chalice), [FastAPI](https://github.com/tiangolo/fastapi), [nox](https://github.com/wntrblm/nox), [pre-commit](https://github.com/pre-commit/pre-commit), [Pyre](https://github.com/facebook/pyre-check), [PyTorch](https://github.com/pytorch/pytorch), [requests](https://github.com/psf/requests))
- `requirements_*.txt` ([TensorFlow](https://github.com/tensorflow/tensorflow))
- none of the above ([Click](https://github.com/pallets/click), [Flask](https://github.com/pallets/flask), [Jinja](https://github.com/pallets/jinja), [sam](https://github.com/aws/serverless-application-model))

Sublime doesn't offer an easy way out; do I have to include every possible permutation here?

File: icons/icons.json [hunk 4]
```diff
@@ -2577,6 +2590,28 @@
...
+	},
+	"file_type_python_typed": {
+		"aliases": [
+			{
+				"base": "source.python",
+				"extensions": [
+					"py.typed"
+				],
+				"name": "Python (PEP 561)",
+				"scope": "source.python.typed"
```
[the remark this answers is not in the document]
Comment (mataha) [reply icons/icons.json:2609]: I've seen [some](https://github.com/python/mypy/blob/master/mypy/py.typed) [projects](https://github.com/vllm-project/vllm/blob/main/vllm/py.typed) [include](https://github.com/cs01/gdbgui/blob/master/gdbgui/py.typed) Python-style comments within, though that's not a rule.

File: icons/icons.json [hunk 5]
```diff
@@ -2577,6 +2590,28 @@
...
+			}
+		],
+		"color": "blue"
+	},
+	"file_type_python_version": {
```
[the remark this answers is not in the document]
Comment (mataha) [reply icons/icons.json]: I agree regarding `.flake8` et al. Is the new definition fine?

**Pull request closed without merging**